openedx/features/smart_referral/api_views.py

```diff
@@ -0,0 +1,31 @@
+from django.http import JsonResponse
+from edx_rest_framework_extensions.auth.session.authentication import SessionAuthenticationAllowInactiveUser
+from rest_framework import status
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.views import APIView
+
+from .helpers import sort_contacts_by_org_and_user_domain, get_platform_contacts_and_non_platform_contacts
+
+
+class FilterContactsAPIView(APIView):
+    """
+    FilterContactsAPIView is used to filter user's contacts based on some specific criteria
+    """
+    authentication_classes = (SessionAuthenticationAllowInactiveUser,)
+    permission_classes = (IsAuthenticated,)
+
+    def post(self, request):
+        user = request.user
+        user_contacts = request.data.get('contacts_list', '')
```

Comment:
because it's a list let's please use, None or `[]` as default value, if use None we need to change implementation of `get_platform_contacts_and_non_platform_contacts` accordingly. but if we use `[]` no change will be required.

Reply:
done